mewarpx/mewarpx/emission.py

```diff
@@ -1461,64 +1559,90 @@ class VolumeEmitter(BaseEmitter):
+            x/y/z/rmin (float): Lower boundary of the volume.
+            x/y/z/rmax (float): Upper boundary of the volume.
         """
         super(VolumeEmitter, self).__init__()
         self.T = T
 
-    def getvoltage(self):
-        """Ideally this is probably the local potential, but default to 0."""
-        return 0.
-
-    def getvoltage_e(self):
-        """Ideally this is probably the local potential, but default to 0."""
-        return self.getvoltage()
-
+        # determine default prism type from simulation geometry
+        self.rectangular = mwxrun.geom_str != 'RZ'
 
-class CartesianVolumeEmitter(VolumeEmitter):
-
-    """VolumeEmitter with bounds specified by x/y/z rectangular prism.
+        # check if a different volume was specified
+        r_bounds_given = (rmin is not None or rmax is not None)
+        xy_bounds_given = (
+            xmin is not None or xmax is not None or ymin is not None or
+            ymax is not None
+        )
 
-    A template ``_get_xv_coords()`` is also provided but can be overridden if
-    needed. It calls ``_get_x_coords()``.
-    """
+        if r_bounds_given and xy_bounds_given:
+            raise AttributeError(
+                "Both rectangular and cylindrical boundaries specified for a "
+                "VolumeEmitter"
+            )
+        if r_bounds_given:
+            self.rectangular = False
+        if xy_bounds_given:
+            self.rectangular = True
 
-    # RZ support should probably take into account the R wall, so we don't list
-    # it as supported.
-    geoms = ['Z', 'XZ', 'XYZ']
+        self.bounds = np.zeros((3, 2))
+        if self.rectangular:
+            for ii, (lim, defaultlim) in enumerate(
+                zip([xmin, xmax, ymin, ymax, zmin, zmax],
+                    [mwxrun.xmin, mwxrun.xmax, mwxrun.ymin,
+                     mwxrun.ymax, mwxrun.zmin, mwxrun.zmax])
+            ):
+                if lim is None:
+                    lim = defaultlim
+                self.bounds[ii // 2, ii % 2] = lim
 
-    def __init__(self, T, xmin=None, xmax=None, ymin=None, ymax=None,
-                 zmin=None, zmax=None):
-        """Initialize emitter boundaries."""
-        super(CartesianVolumeEmitter, self).__init__(T=T)
+            self.volume = np.prod(self.bounds[:, 1] - self.bounds[:, 0])
 
-        self.bounds = np.zeros((3, 2))
-        for ii, (lim, defaultlim) in enumerate(
-            zip([xmin, xmax, ymin, ymax, zmin, zmax],
-                [mwxrun.xmin, mwxrun.xmax, mwxrun.ymin,
-                 mwxrun.ymax, mwxrun.zmin, mwxrun.zmax])
-        ):
-            if lim is None:
-                lim = defaultlim
-            self.bounds[ii // 2, ii % 2] = lim
+        else:
+            for ii, (lim, defaultlim) in enumerate(
+                zip([rmin, rmax, 0, 2.0*np.pi, zmin, zmax],
+                    [mwxrun.xmin, mwxrun.xmax, mwxrun.ymin,
+                     mwxrun.ymax, mwxrun.zmin, mwxrun.zmax])
```

Comment:
Need to change defaults I think.

Reply:
Seeing as in `mwxrun` we set `self.xmin = self.ymin = self.rmin = self.grid.rmin` and same for `xmax` it is will work fine as is but happy to change it to make things clearer.